### PR TITLE
Issue 20688 property update site name

### PR DESCRIPTION
**Pull request by erickgonzalez**
Check if the property `forceExecution` is set to true, if so the user can update the siteName without issues. If not sent, we should throw an **illegalArgumentException** (aka 400) if the user is trying to update the siteName.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java [hunk 1]
```diff
@@ -412,41 +413,47 @@ public Host find(final Contentlet contentlet,
...
             contentletHost = new Contentlet();
             contentletHost.setStructureInode(hostType().inode() );
         }
 
-        contentletHost.getMap().put(Contentlet.DONT_VALIDATE_ME, host.getMap().get(Contentlet.DONT_VALIDATE_ME));
-        APILocator.getContentletAPI().copyProperties(contentletHost, host.getMap());
+        if(null != contentletHost.get(Host.HOST_NAME_KEY) && null != hostToBeSaved.get(Host.HOST_NAME_KEY) &&
+                !contentletHost.get(Host.HOST_NAME_KEY).equals(hostToBeSaved.get(Host.HOST_NAME_KEY)) &&
+                !hostToBeSaved.getBoolProperty("forceExecution")){
+            throw new BadRequestException("Updating the hostName is a Dangerous Execution, to achieve this 'forceExecution': true property needs to be sent.");
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java:433]: We should not throw a BadRequestException here

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java [hunk 2]
```diff
@@ -412,41 +413,47 @@ public Host find(final Contentlet contentlet,
...
 
             contentletHost = new Contentlet();
             contentletHost.setStructureInode(hostType().inode() );
         }
 
-        contentletHost.getMap().put(Contentlet.DONT_VALIDATE_ME, host.getMap().get(Contentlet.DONT_VALIDATE_ME));
-        APILocator.getContentletAPI().copyProperties(contentletHost, host.getMap());
+        if(null != contentletHost.get(Host.HOST_NAME_KEY) && null != hostToBeSaved.get(Host.HOST_NAME_KEY) &&
+                !contentletHost.get(Host.HOST_NAME_KEY).equals(hostToBeSaved.get(Host.HOST_NAME_KEY)) &&
+                !hostToBeSaved.getBoolProperty("forceExecution")){
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java:432]: I am not sure if this property should be here, maybe we should check it in the end point and then call another API method something like saveAndForceExecution (maybe)

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java (hunk 1 above)
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java:433]: BadRequestException is an exception only for rest level, at API level would be more convenient:
IllegalArgumentException

Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java:433]: done

**Review by jdotcms — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

**Pull request merged**